crates/forge/tests/cli/script.rs

```diff
@@ -855,6 +855,46 @@ forgetest_async!(can_deploy_with_create2, |prj, cmd| {
         .run(ScriptOutcome::ScriptFailed);
 });
 
+forgetest_async!(can_deploy_with_custom_create2, |prj, cmd| {
+    let (api, handle) = spawn(NodeConfig::test()).await;
+    let mut tester = ScriptTester::new_broadcast(cmd, &handle.http_endpoint(), prj.root());
+    let create2 = Address::from_str("0x0000000000000000000000000000000000b4956c").unwrap();
+
+    // Prepare CREATE2 Deployer
+    api.anvil_set_code(
+        create2,
+        Bytes::from_static(foundry_evm::constants::DEFAULT_CREATE2_DEPLOYER_RUNTIME_CODE),
+    )
+    .await
+    .unwrap();
+
+    tester
+        .add_deployer(0)
+        .load_private_keys(&[0])
+        .await
+        .add_create2_deployer(create2)
+        .add_sig("BroadcastTestNoLinking", "deployCreate2()")
+        .simulate(ScriptOutcome::OkSimulation)
+        .broadcast(ScriptOutcome::OkBroadcast)
+        .assert_nonce_increment(&[(0, 2)])
+        .await;
```

Comment:
can we assert here that it was deployed to the right address? either need a snapbox fixture or a `cast.get_code` check

Reply:
sounds reasonable

Reply:
I think we should also add a test to check the create2's code is the same as the default one.

Reply:
still working on the deployed address test

Reply:
@klkvr sorry, can't think of a good way to test it with deployed contrat, so I test it by compare the contract address directly in the script in https://github.com/foundry-rs/foundry/blob/64545738dd183c64bd8d7a4ec94f7ba03c8d8d05/testdata/default/cheats/Broadcast.t.sol#L222-L243

Reply:
you could you anvil's handle `api` https://github.com/jsvisa/foundry/blob/64545738dd183c64bd8d7a4ec94f7ba03c8d8d05/crates/forge/tests/cli/script.rs#L885

and then just do something like `assert!(!api.get_code(...).is_empty());

but it's fine anyway :)

Reply:
Just a quick question, if we use anvil to get the contract code, but how could we get the contract address? I was thought of calculate the address by
1. compiling the solidity to get the init code
2. get the initcodehash + salt + deployer address

but stuck in the first step :sob:

Reply:
I guess you could use the anvil api to get the receipt and contract address from there, smth like we're getting the hash of last tx here https://github.com/foundry-rs/foundry/blob/7f41280ee071193557f73f16bae9aee9a5548ee8/crates/cast/tests/cli/main.rs#L1745